pkg/pod/status.go

```diff
@@ -273,8 +273,20 @@ func filterResults(results []result.RunResult, specResults []v1.TaskResult) ([]v
 	var taskResults []v1.TaskRunResult
 	var filteredResults []result.RunResult
 	neededTypes := make(map[string]v1.ResultsType)
+	neededStepTypes := make(map[string]v1.ResultsType)
+	neededStepResults := make(map[string]string)
 	for _, r := range specResults {
 		neededTypes[r.Name] = r.Type
+		if r.Value != nil {
+			if r.Value.StringVal != "" {
+				stepName, resultName, err := v1.ExtractStepResultName(r.Value.StringVal)
+				if err != nil {
+					continue
+				}
+				neededStepResults[fmt.Sprintf("%s.%s", stepName, resultName)] = r.Name
+				neededStepTypes[fmt.Sprintf("%s.%s", stepName, resultName)] = r.Type
+			}
+		}
```

Comment:
I don't think this is quite what you want - this func is filtering values based on the values included in the TaskSpec. 

e.g. as is this is only going to preserve values that Tasks chose to export, and throw away all other Step results, even if they are declared in the Step.

What you probably want to do is take in the task results + step results, then return the filtered Task results, filtered Step results, and everything else, then handle the interleaving / value resolution later.